### ISSUE TITLE
Add a mode that doesn't delete all apigateway resources

### ISSUE THREAD

**Issue by corespace** [awslabs/chalice#123]
Hi all,

So far I'm loving chalice, but I've got a use case where my POST workflow is super-complicated (and therefore needs some python code) but my GET workflow can just be a pass-thru to S3. Chalice deletes my resources every time it deploys, which is quite obnoxious when all I really need it to do is update the lambda code and not the apigateway mappings. Any love for such a feature? Let me know if I can/should help implement!

Thanks,
Kyle

**Comment by jamesls** [comment]
Seems reasonable, this will require some design discussion.  The ideal solution is that we keep track of what resources we create on deploy, and on re-deploy we only remove the resources we've created.  However, that's going to be possible if you deploy on one machine and then want to push an update on another machine.  Open to other implementation ideas.

Marking as a feature request.

**Comment by corespace** [comment]
Yeah I'm not as familiar with apigateway as you are, I'm sure. Do you have time to explain why all the resources need to be deleted for chalice to work? It seems to me that really we'd just need to delete the method / method integration... unless your point is that you cannot tell which resources/methods/integrations should stick around and which ones are cruft from a previous version of what the customer was working on. Perhaps a better solution here is to allow the customer to specify some configuration for additional resources/methods/integrations they want chalice to manage? Mine would look something like this:

resources:
/things
/things/{id}

methods:
/things POST (managed via code & chalice)
/things/{id} GET (pass thru to S3)

**Comment by corespace** [comment]
Actually I think the minimally viable solution to my problem would be a way to run chalice without doing the deploy to the dev stage. The additional python script I've written should add the additional resources/integrations and then do the deploy. Does that make this easier?
